tests/integration/targets/ndo_interface_setting/tasks/main.yml

```diff
@@ -0,0 +1,721 @@
+# Test code for the MSO modules
+# Copyright: (c) 2024, Anvitha Jain (@anvjain) <anvjain@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI MultiSite host, username and password
+  ansible.builtin.fail:
+    msg: 'Please define the following variables: mso_hostname, mso_username and mso_password.'
+  when: mso_hostname is not defined or mso_username is not defined or mso_password is not defined
+
+# CLEAN ENVIRONMENT
+- name: Set vars
+  ansible.builtin.set_fact:
+    mso_info: &mso_info
+      host: '{{ mso_hostname }}'
+      username: '{{ mso_username }}'
+      password: '{{ mso_password }}'
+      validate_certs: '{{ mso_validate_certs | default(false) }}'
+      use_ssl: '{{ mso_use_ssl | default(true) }}'
+      use_proxy: '{{ mso_use_proxy | default(true) }}'
+      output_level: '{{ mso_output_level | default("debug") }}'
+
+# QUERY VERSION
+- name: Query MSO version
+  cisco.mso.mso_version:
+    <<: *mso_info
+    state: query
+  register: version
+
+
+- name: Execute tasks only for MSO version > 4.3
+  when: version.current.version is version('4.3', '>=')
+  block:
+    - name: Remove fabric template
+      cisco.mso.ndo_template: &template_absent
+        <<: *mso_info
+        name: ansible_fabric_policy_template
+        type: fabric_policy
+        state: absent
+
+    - name: Create a fabric template
+      cisco.mso.ndo_template:
+        <<: *template_absent
+        state: present
+
+    # Create domains
+    - name: Create a physical domain
+      cisco.mso.ndo_physical_domain:
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        name: '{{ item }}'
+        state: present
+      loop:
+        - 'ansible_physical_domain'
+        - 'ansible_physical_domain_2'
+
+    - name: Create a l3 domain
+      cisco.mso.ndo_l3_domain:
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        name: '{{ item }}'
+        state: present
+      loop:
+        - 'ansible_l3_domain'
+        - 'ansible_l3_domain_2'
+
+    #  Create SyncE Interface Policy
+    - name: Create a SyncE Interface Policy
+      cisco.mso.ndo_synce_interface_policy:
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        interface_policy: '{{ item }}'
+        state: present
+      loop:
+        - 'ansible_synce_interface_policy'
+        - 'ansible_synce_interface_policy_2'
+
+    # # Create MACSec Policy
+    # - name: Create a MACSec Policy
+    #   cisco.mso.ndo_macsec_policy:
+    #     <<: *mso_info
+    #     template: ansible_fabric_policy_template
+    #     macsec_policy: '{{ item }}'
+    #     interface_type: access
+    #     state: present
+    #   loop:
+    #     - 'ansible_macsec_policy'
+    #     - 'ansible_macsec_policy_2'
```

Comment:
if not used, we should remove it from test

Reply:
it is waiting for MACSec module